.github/workflows/ci.yml

```diff
@@ -8,24 +8,38 @@ jobs:
   docker:
     name: Build and test docker image
 
+    strategy:
+      matrix:
+        nominatim:
+          - version: 3.7
+            update_command: docker exec -i nominatim sudo -u nominatim nominatim replication --project-dir /nominatim --once
+          - version: 3.6
+            update_command: docker exec -i nominatim sudo -u nominatim ./src/build/utils/update.php
+
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
 
       - name: Build docker image
         run: docker build . -t nominatim
-        working-directory: 3.6
+        working-directory: ${{ matrix.nominatim.version }}
 
-      - name: Check import without mounts
+      - name: Check import and update
         working-directory: .github/workflows
         run: |-
+          # get the first day of the previous month to make sure that there really are updates to apply
+          # (Geofabrik stores a snapshot from the first of the month for up to 3 months)
+          first_of_previous_month=`date --date="1 months ago" +%y%m01`
           docker run -i --rm \
-            -e PBF_URL=http://download.geofabrik.de/europe/monaco-latest.osm.pbf \
+            -e PBF_URL=http://download.geofabrik.de/europe/monaco-${first_of_previous_month}.osm.pbf \
             -e REPLICATION_URL=http://download.geofabrik.de/europe/monaco-updates/ \
             -p 8080:8080 \
+            --name nominatim \
             nominatim &
-          sleep 45
+          sleep 35
```

Comment:
AFAIK 35 seconds is not enough, see: https://github.com/mediagis/nominatim-docker/runs/2507636802?check_suite_focus=true#step:4:360

If the download is not fast enough we may run into problems.

Reply:
The script to check the response has a retry mechanism with an exponential backoff. It just retries for up to ~ a minute before giving up. In my view that is safe enough.